### PR TITLE
added rosdep rule for fastnumbers pip package

### PR DESCRIPTION
**Pull request by niomate**
Added rosdep rule "python-fastnumbers-pip" for Arch, Debian, Fedora, OSX and Ubuntu. Passes all unit tests on my machine. Additionally ran "scripts/clean_rosdep_yaml.py" for as suggested by the tests.

### CODE REVIEW

File: rosdep/python.yaml [hunk 1]
```diff
@@ -207,7 +207,7 @@ jupyter-nbconvert:
   gentoo: [dev-python/nbconvert]
   ubuntu:
     '*': [jupyter-nbconvert]
-    xenial: null
+    xenial:
```
Comment (wjwwood) [inline rosdep/python.yaml:210]: This change looks unrelated to your pull request. Please remove this part of the diff.

File: rosdep/python.yaml [hunk 2]
```diff
@@ -731,7 +731,7 @@ python-boto3:
   opensuse: [python-boto3]
   ubuntu:
     '*': [python-boto3]
-    trusty: null
```
Comment (wjwwood) [inline rosdep/python.yaml:734]: Same here.

**Review by wjwwood — CHANGES_REQUESTED** [review]
There are a lot of changes that are unrelated to your new rosdep rule please remove those changes.

File: rosdep/python.yaml (hunk 2 above)
Comment (Ramdambo) [reply rosdep/python.yaml:734]: Oh yes, did not see that, must have been removed by clean_rosdep_yaml.py. I'll fix it!

File: rosdep/python.yaml [hunk 3]
```diff
@@ -5676,7 +5692,7 @@ python3-flake8:
       packages: [flake8]
   rhel:
     '*': ['python%{python3_pkgversion}-flake8']
-    '7': null
+    '7':
```
Comment (wjwwood) [inline rosdep/python.yaml:5695]: There are still some unwanted changes here. Even if I were to agree with some of them, they should be done as part of another pull request. Please get the diff down to just your new stanza for the new rosdep rule, thanks :)

Comment (tfoote) [reply rosdep/python.yaml:5695]: We specifically use null to indicate that it's unmet. 

From: https://yaml.org/type/null.html
> Note that a null is different from an empty string and that a mapping entry with some key and a null value is valid and different from not having that key in the mapping.

**Review by tfoote — CHANGES_REQUESTED** [review]

File: rosdep/python.yaml (hunk 3 above)
Comment (Ramdambo) [reply rosdep/python.yaml]: Ah ok. Maybe the clean_rosdep_yaml script should be updated then ? Because when I ran it, it removed all the null values.

**Comment by Ramdambo** [comment]
Should be all reverted now again.

File: rosdep/python.yaml (hunk 3 above)
Comment (clalancette) [reply rosdep/python.yaml]: > Ah ok. Maybe the clean_rosdep_yaml script should be updated then ? Because when I ran it, it removed all the null values. Or are you actually not supposed to run this script ? Because an error message in the tests suggested to run it.

Ah, that script may be out-of-date.  I don't know that its been updated to handle the `null` values; those are somewhat new.  If you'd like to contribute a PR to update it, we'd be happy to review!

File: rosdep/python.yaml [hunk 4]
```diff
@@ -2098,7 +2114,7 @@ python-imaging:
   arch: [python2-pillow]
   debian:
     '*': [python-pil]
-    jessie:  [python-imaging]
+    jessie: [python-imaging]
```
Comment (clalancette) [inline rosdep/python.yaml:2117]: This change (and the ones below) still don't belong in this PR.  I actually think they are valid fixes, but please submit them separately.  This PR should just be focused on the `fastnumbers` package.

**Review by clalancette — CHANGES_REQUESTED** [review]

File: rosdep/python.yaml (hunk 3 above)
Comment (Ramdambo) [reply rosdep/python.yaml]: Ok, I'll have a look once I get the chance!

File: rosdep/python.yaml [hunk 5]
```diff
@@ -1513,6 +1513,22 @@ python-fastdtw-pip:
   ubuntu:
     pip:
       packages: [python-fastdtw]
+python-fastnumbers-pip:
```
Comment (clalancette) [inline rosdep/python.yaml:1516]: Just to be clear; this is implicitly a package for Python 2.  Are you using it for Python 2?  If so, this is OK as-is.  If not, I'd prefer to make this:

```
python3-fastnumbers-pip
```

Comment (Ramdambo) [reply rosdep/python.yaml:1516]: Good point, I oversaw that completely. I was using it with Python 2.7 and it seems to work just fine, but I double checked the documentation on PyPi because I was not completely sure anymore whether 2.7 is actually supported and turns out it's for versions >= 3.5.
I'll update it ASAP !

File: rosdep/python.yaml [hunk 6]
```diff
@@ -1513,6 +1513,22 @@ python-fastdtw-pip:
   ubuntu:
     pip:
       packages: [python-fastdtw]
+python3-fastnumbers-pip:
```
Comment (clalancette) [inline rosdep/python.yaml:1516]: This is failing CI because the whole section has to move down into the `python3-` section.

Comment (Ramdambo) [reply rosdep/python.yaml:1516]: Oooooh, I was wondering why it failed. Sorry, I overread the section on `python-` vs `python3-` placements.

**Review by jacobperron — APPROVED** [review]

**Review by ivanpauno — APPROVED** [review]
